File: LeetCode/Python/0345_Reverse_Vowels_of_a_String.py
# 345. Reverse Vowels of a String
# https://leetcode.com/problems/reverse-vowels-of-a-string/


class Solution(object):
    def reverseVowels(self, s):
        vowels = ["a", "e", "i", "o", "u", "A", "E", "I", "O", "U"]
        vow = ""
        for char in s:
            if char in vowels:
                vow += char
        vow = vow[::-1]

        new_str = ""
        i = 0
        for char in s:
            if char in vowels:
                new_str += vow[i]
                i += 1
            else:
                new_str += char

        return new_str

File: LeetCode/JavaScript/0345-Reverse-Vowels-of-a-String.js
// 345. Reverse Vowels of a String
// https://leetcode.com/problems/reverse-vowels-of-a-string/description/

/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function (s) {
  const vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];
  const vowRes = [];
  let res = '';
  for (const char of s) if (vowels.includes(char)) vowRes.push(char);
  for (let i = 0; i < s.length; i++) {
    if (vowels.includes(s[i])) res += vowRes.pop();
    else res += s[i];
  }
  return res;
};
